File: lab6/matrixWithoutErrors.cpp
#include <iostream>
using namespace std;
int main() {
	cout << "Welcome to Matrix Multiply:" << endl;
	cout << "Enter the size of the Matrix: " << endl;

	int size = 0;
	cin >> size;

	int row = size;
	int col = size;

	int ** A = new int *[row];
	int ** B = new int *[row];
	int ** C = new int *[row];

	for (int i = 0; i<row; i++)
	{
		A[i] = new int[size];
		B[i] = new int[size];
		C[i] = new int[size];
	}

	for (int i = 0; i<row; i++)
	{
		for (int j = 0; j<col; j++)
		{
			A[i][j] = 0;
			cout << "Enter value for Position " << "A[" << i << "]" << "[" << j << "]" << endl;
			cin >> A[i][j];
		}
	}

	for (int i = 0; i<row; i++)
	{
		for (int j = 0; j<col; j++)
		{
			B[i][j] = 0;
			C[i][j] = 0;
			cout << "Enter value for Position " << "B[" << i << "]" << "[" << j << "]" << endl;
			cin >> B[i][j];
		}
	}


	for (int row1 = 0; row1<size; row1++)
	{
		for (int j = 0; j<col; j++)
		{
			for (int row2 = 0; row2 < col; row2++)
			{
				
				cout << "the values being multiplied are: " << A[row1][row2] << " and " << B[row2][j] << endl;
				C[row1][j] += (A[row1][row2] * B[row2][j]);
				/*cout << " C[" << row1<< "][" << j << "] is: " << C[row1][j] << endl;*/
			}
		}
	}

	for (int i = 0; i<size; i++)
	{
		for (int j = 0; j<size; j++)
		{
			cout<<"here"<<endl;
			cout << " C[" << i << "][" << j << "] is: " << C[i][j] << endl;
		}
	}

	for (int i = 0; i< row; i++)
	{
		delete[] A[i];
		delete[] B[i];
		delete[] C[i];
	}

}

File: lab7/matrixwithoutErrors.py
import sys


def matrixMultiply():
    print("Welcome to Matrix Multiply")
    inputFromUser= input("Enter the size of the Matrix")
    
    size = int(inputFromUser)
    mat1 = [[0]*5 for i in range(size)]
    mat2 = [[0]*5 for i in range(size)]
    ans = [[0]*5 for i in range(size)]
    
    for i in range(size):
        for j in range(size):
            inputMSG = "Enter the value for position A["+str(i)+"]["+str(j)+"]: "
            inputForMatrixValue = input(inputMSG)
            value = int(inputForMatrixValue)
            mat1[i][j] = value

    for i in range(size):
        for j in range(size):
            inputMSG = "Enter the value for position B["+str(i)+"]["+str(j)+"]: "
            inputForMatrixValue = input(inputMSG)    
            value = int(inputForMatrixValue)
            mat2[i][j] = value

    for row in range(size):
        for column in range(size):
            for row2 in range(size):
                ans[row][column] += mat1[row][row2]*mat2[row2][column]
    
    for i in range(size):
        for j in range(size):
            print("element ans["+str(i)+"]["+str(j)+"]" + str(ans[i][j]))
	
matrixMultiply();

